documentation/add/04_mpi_framework.md

```diff
@@ -109,185 +146,232 @@ The main methods and properties (M/P) of **JobExecutor** involved in a job contr
 
-Different purposes can be best served using different types of messages. There are 3 purposes, differentiated in Horace.
+Different operations can be handled using different types of messages. There are 3 main types, differentiated in Horace.
 
-1. logging, diagnostics and informing user about progress of a job.
+1. Logging, diagnostics and informing user about progress of a job.
 2. Interprocess data exchange, necessary for performing a particular user job and
-3. The information about error or failure.
+3. Information about error or failure.
 
 
-Correspondent types of messages created to serve these purposes:
+Corresponding types of messages created to serve these purposes:
 
-1. Non-blocking (transient)  messages. Informing user about a job progress is important task, but user is interested only in the final value. Different independent task should not be synchronous with regard to log messages as only some average progress is required. Non-blocking messages are best suited for such kind of task. If more then one non-blocking message is present in the MPI messages queue, the client receives only the last message.
-2. Blocking messages are necessary to transfer information between independent workers when the result of calculations on a single node depends on the results on other nodes.
-3. Persistent messages necessary to indicate some critical conditions or error states, as, for example, parallel interrupts available in MPI are not available in file-based messages system (*ClusterHerbert*) and not accessible for Matlab MPI (*ClusterParpool*). The persistent messages used to carry out such information. 
+1. Non-blocking (transient)  messages. Informing user about a job progress is an important task, but often the user is interested only in the final value.
+   Different independent tasks should not be synchronous with regard to log messages as only some average progress is required. 
+   Non-blocking messages are best suited for such kind of task. 
+   If more then one non-blocking message is present in the MPI messages queue, the client receives only the last message.
+3. Blocking messages are necessary to transfer information between independent workers when the result of calculations on a single node depends on the results on other nodes.
+4. Persistent messages necessary to indicate some critical conditions or error states, as, for example, 
+   parallel interrupts available in MPI are not available in file-based messages system (*ClusterHerbert*) and not accessible for Matlab MPI (*ClusterParpool*). 
+   Persistent messages are used to carry such information. 
 
-To provide the described flexibility and variability of messages, all messages used by Horace MPI are children of **aMessage**  class and subscribed to a messages factory (see below). 
+To provide the described flexibility and variability of messages, all messages used by Horace MPI are children of **aMessage** class and subscribed to a messages factory (see below). 
```

Comment:
This is MPI  specific. Wht happens with the other frameworks?

Reply:
its not MPI specific -- all frameworks communicate using the same messages. Probably word Horace MPI here is confusing -- it points to any parallel framework used by Horace.